onnx/defs/nn/defs.cc

```diff
@@ -1534,37 +1534,49 @@ ONNX_OPERATOR_SET_SCHEMA(
             "var",
             "running (training) or estimated (testing) variance tensor of shape (C).",
             "T")
+        .Input(
+            5,
+            "training_mode",
+            "If set to true, run spatial batch normalization in training mode, default is false.",
+            "T1",
+            OpSchema::Optional)
         .Output(0, "Y", "The output tensor of the same shape as X", "T")
         .Output(
             1,
             "mean",
-            "The running mean after the BatchNormalization operator.",
+            "The running mean after the BatchNormalization operator."
+            "This output is populated only when training_mode is set to true.",
             "T",
             OpSchema::Optional)
         .Output(
             2,
             "var",
-            "The running variance after the BatchNormalization operator.",
+            "The running variance after the BatchNormalization operator."
+            "This output is populated only when training_mode is set to true.",
             "T",
             OpSchema::Optional)
         .Output(
             3,
             "saved_mean",
-            "Saved mean used during training to speed up gradient "
-            "computation.",
+            "Saved mean used during training to speed up gradient computation."
+            "This output is populated only when training_mode is set to true.",
             "T",
             OpSchema::Optional)
         .Output(
             4,
             "saved_var",
-            "Saved variance used during training to speed up "
-            "gradient computation.",
+            "Saved variance used during training to speed up gradient computation."
+            "This output is populated only when training_mode is set to true.",
             "T",
             OpSchema::Optional)
         .TypeConstraint(
             "T",
             {"tensor(float16)", "tensor(float)", "tensor(double)"},
             "Constrain input and output types to float tensors.")
+        .TypeConstraint(
+            "T1",
+            {"tensor(bool)"},
+            "Constrain input 'training_mode' types to boolean tensors.")
```

Comment:
If we really make `mode` an input, we need to check if it's a Boolean scalar in the shape inference below.